src/status_im2/common/animated_header_list/view.cljs

```diff
@@ -0,0 +1,101 @@
+(ns status-im2.common.animated-header-list.view
+  (:require
+   [quo2.core :as quo]
+   [react-native.core :as rn]
+   [react-native.platform :as platform]
+   [react-native.reanimated :as reanimated]
+   [react-native.safe-area :as safe-area]
+   [reagent.core :as reagent]
+   [quo2.foundations.colors :as colors]
+   [status-im.ui.components.fast-image :as fast-image]
+   [status-im2.common.animated-header-list.style :as style]
+   [oops.core :as oops]
+   [utils.re-frame :as rf]))
+
+(def header-height 192)
+
+(def threshold (/ header-height 2))
+
+(defn scroll-handler
+  [event initial-y scroll-y opacity-value]
+  (let [current-y (- (oops/oget event "nativeEvent.contentOffset.y") initial-y)]
+    (reanimated/set-shared-value scroll-y current-y)
+    (reanimated/set-shared-value opacity-value
+                                 (reanimated/with-timing (if (> current-y threshold) 1 0)))))
+
+(defn header
+  [{:keys [theme-color display-picture-comp cover-uri]} top-inset scroll-y]
+  (let [input-range     [0 threshold]
+        output-range    [1 0.4]
+        scale-animation (reanimated/interpolate scroll-y
+                                                input-range
+                                                output-range
+                                                {:extrapolateLeft  "clamp"
+                                                 :extrapolateRight "clamp"})]
+    [rn/view
+     {:style {:height           header-height
+              :background-color (or theme-color (colors/theme-colors colors/white colors/neutral-95))
+              :margin-top       (when platform/ios? (- top-inset))}}
+     (when cover-uri
+       [fast-image/fast-image
+        {:style  {:width  "100%"
+                  :height "100%"}
+         :source {:uri cover-uri}}])
+     [rn/view {:style style/header-bottom-part}]
+     [reanimated/view {:style (style/entity-picture scale-animation)}
+      [display-picture-comp]]]))
+
+(defn animated-header-list
+  [{:keys [title-comp main-comp] :as parameters}]
+  [safe-area/consumer
+   (fn [insets]
+     (let [window-height     (:height (rn/get-window))
+           status-bar-height (rn/status-bar-height)
+           bottom-inset      (:bottom insets)
+           ;; view height calculation is different because window height is different on iOS and Android:
+           ;; https://i.stack.imgur.com/LSyW5.png
```

Comment:
[Nitpick] Imgur is not a reliable source of information. Something like a git repository, Wikipedia or SO would be preferable.